src/Controller/DisplayPaymentFailedPageAction.php

```diff
@@ -22,12 +23,24 @@ public function __construct(
     public function __invoke(Request $request, string $orderToken): Response
     {
         $order = $this->findOrderOr404($orderToken);
-        $payment = $order->getLastPayment();
+        $newPayment = $order->getLastPayment();
+        $failedPayment = $order->getPayments()->get($order->getPayments()->count() - 2);
```

Comment:
What kind of sorcery is it?

Reply:
I forgot to refactor this, but the main idea should stay I believe. We need the error message from the failed payment. Right now, the last payment is a new payment, not the one that failed. That’s why I take one more payment behind.

Reply:
we can tak last failed payment, as there is option to pass argument to that function

Reply:
fixed